LessonOneFinalModuleHW/Program.cs

```diff
@@ -0,0 +1,245 @@
+                        string userTaskDelete = Console.ReadLine();
+
+                        if (userTaskDelete == "*")
+                        {
+                            Array.Clear(taskArray);
+                            Array.Clear(finishedTaskArray);
+                            Array.Clear(datesOfFinishedTaskArray);
+
+                            Console.WriteLine("To-Do list is cleaned.");
+                        }
+                        else
+                        {
+                            (taskArray, finishedTaskArray, datesOfFinishedTaskArray) = DeleteItemFunc(taskArray, finishedTaskArray, datesOfFinishedTaskArray, userTaskDelete);
+                        }
+
+                        continue;
+                    case "show":
+                        Console.WriteLine("Sub-command:\n1 - task done\n0 - task not done");
+                        string userCmdLine = Console.ReadLine();
+                        PrintArray(taskArray, finishedTaskArray, datesOfFinishedTaskArray, userCmdLine);
+                        continue;
+                    case "mark-as":
+                        Console.WriteLine("Sub-command:\n1 - task done\n0 - task not done");
+                        string userStatus = Console.ReadLine();
+                        Console.WriteLine("Please write task-item:");
+                        string userTaskLine = Console.ReadLine();
+
+                        (taskArray, finishedTaskArray, datesOfFinishedTaskArray) =
+                            MarkAsFunc(taskArray, finishedTaskArray, datesOfFinishedTaskArray, userTaskLine, userStatus);
+                        continue;
+                }
+            }
+        }
+
+        static bool ExitFunc(bool value)
+        {
+            Console.WriteLine("Exiting program");
+            value = false;
+            return value;
+        }
+
+        static string[] AddItemFunc(string[] array, string value)
+        {
+            if (value == String.Empty)
+            {
+                Console.WriteLine("You're task-item is empty!");
+            }
+
+            else
+            {
+                array = Array.ConvertAll(array, x => x.ToLower()).ToArray();
+
+                value = value.ToLower();
+
+                string valueEmpty = Array.Find(array, i => i.StartsWith(value, StringComparison.Ordinal));
+
+                if (valueEmpty != null && !DateTime.TryParse(value, out DateTime Temp))
+                {
+                    Console.WriteLine("You're task-item is already there!");
+                }
+                else
+                {
+                    array = array.Append(value).ToArray();
+                    Console.WriteLine($"You're task-item '{value}' was added!");
+                }
+            }
+
+            return array;
+        }
+
+        static (string[] taskArray, string[] finishedTaskArray, string[] dateArray) DeleteItemFunc(string[] taskArray, string[] finishedTaskArray, string[] dateArray, string value)
```

Comment:
What is the point of returning new arrays? You have a 'ref' modifier to repoint the existing array to the new memory address.